### PR TITLE
Pylint fixes

### PR DESCRIPTION
**Pull request by MuellerSeb**
... I've done it. Pylint is a pain, but a good one.

I created a setup for Pylint in `pyproject.toml` that should warn us about bad practice. I added a new stage in CI to check pylint.

Certain Pylint rules are disabled per file, where it makes sence (at least for me).

It helped me figure out all of the circular imports (see #24). Especially the matplotlib imports are now in a try block and later on deferred.

All remaining pylint warnings were sorted out here.

### CODE REVIEW

**Comment by MuellerSeb** [comment]
@LSchueler this is ready for review.

File: gstools/field/base.py [hunk 1]
```diff
@@ -297,9 +296,8 @@ def plot(
                 "Specify 'scalar' or 'vector' before plotting."
             )
 
-        elif self.value_type == "scalar":
+        if self.value_type == "scalar":
```
Comment (LSchueler) [inline gstools/field/base.py:299]: Is some super deep Python knowledge required to understand why elif is wrong here? Or is this a convention to separate error raising from other branching logic?

**Review by LSchueler — COMMENTED** [review]
I just had a very very brief look at this PR. Some changes I could not comprehend so quickly, but most of them look pretty harmless.
Maybe it would be helpful to have one central file (some source code file) where the pylint disabling codes are summarized or has the documentaion gotten better over time? - The last time I was setting the linter, it was a real pain to find the correct codes.
Actually, I think I would prefer a description on why each setting was made for each file. I'm just not sure how cluttered the files would be.

But anyway, I'm really curious to see what the Linter will complain about in my future code :-)

File: gstools/field/base.py (hunk 1 above)
Comment (MuellerSeb) [reply gstools/field/base.py:299]: See here: https://vald-phoenix.github.io/pylint-errors/plerr/errors/refactoring/R1720

Comment (MuellerSeb) [reply gstools/field/base.py:299]: The rule simply says: no need for an `else` after a `raise`.

**Comment by MuellerSeb** [comment]
@LSchueler there is this Project: https://vald-phoenix.github.io/pylint-errors/ where the error-codes are documented quite well. I added a section in `CONTRIBUTING.md` about the occasionally disabled pylint checks. Hope you like it.

**Comment by LSchueler** [comment]
> @LSchueler there is this Project: https://vald-phoenix.github.io/pylint-errors/ where the error-codes are documented quite well. I added a section in `CONTRIBUTING.md` about the occasionally disabled pylint checks. Hope you like it.

That's great, I like the summary and also that you can find it in `CONTRIBUTING.md`.

**Review by LSchueler — APPROVED** [review]

**Pull request merged**